java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java

```diff
@@ -66,8 +81,62 @@ private static boolean isObjectMethod(Symbol.MethodSymbol method) {
   }
 
   private static boolean isAnnotatedOverride(Symbol.MethodSymbol method) {
-    return method.metadata()
-      .isAnnotatedWith("java.lang.Override");
+    return method.metadata().isAnnotatedWith("java.lang.Override");
+  }
+
+  /**
+   * Place the @Override annotation as first annotation, on top of the signature
+   * @param methodTree the method to annotate
+   * @return the quick-fix adding the @Override annotation one line above the signature
+   */
+  private JavaQuickFix quickFix(MethodTree methodTree) {
+    ModifiersTree modifiersTree = methodTree.modifiers();
+    Tree targetTree = modifiersTree.isEmpty() ? QuickFixHelper.nextToken(modifiersTree) : modifiersTree.get(0);
+    return JavaQuickFix
+      .newQuickFix("Add \"@Override\" annotation")
+      .addTextEdit(JavaTextEdit.insertBeforeTree(targetTree, "@Override" + newLineWithPadding(targetTree)))
+      .build();
+  }
+
+  private String newLineWithPadding(Tree tree) {
+    String newLineSequence = findNewLineSequence();
+
+    SyntaxToken firstToken = tree.firstToken();
+    String startOfLine = context.getFileLines()
+      .get(firstToken.line() - 1)
+      .substring(0, firstToken.column());
+
+    if (startOfLine.trim().isEmpty()) {
+      // only whitespace characters
+      return newLineSequence + startOfLine;
+    }
+
+    Matcher paddingMatcher = PADDING_PATTERN.matcher(startOfLine);
+    if (paddingMatcher.find()) {
+      // use same spacing as the current line, potentially including mixed tabs and spaces
+      return newLineSequence + paddingMatcher.group(1);
+    }
+
+    // no whitespaces?
+    return newLineSequence;
```

Comment:
In practice
```
  static class InnerClassSingleLine extends OverrideAnnotationCheckSimple { public void get() { } }
```
becomes (in my IDE)
```
  static class InnerClassSingleLine extends OverrideAnnotationCheckSimple {
    @Override
  public void get() { } }
```



If the method contains something before at the same line, we are already talking about strange formatted code, I don't think we should try to do something smart to cleanly add the code, a simple space (even without a new line) seems enough.